nautobot/extras/forms.py

```diff
@@ -629,13 +631,45 @@ class Meta:
 class CustomLinkFilterForm(BootstrapMixin, forms.Form):
     model = CustomLink
     q = forms.CharField(required=False, label="Search")
-    content_type = forms.ModelChoiceField(
+    content_type = CSVContentTypeField(
         queryset=ContentType.objects.filter(FeatureQuery("custom_links").get_query()).order_by("app_label", "model"),
         required=False,
         label="Content Type",
     )
 
 
+#
+# Dynamic Groups
+#
+
+
+class DynamicGroupForm(BootstrapMixin, CustomFieldModelForm, RelationshipModelForm):
```

Comment:
Similarly, this should ideally be 

```suggestion
class DynamicGroupForm(NautobotModelForm):
```

using the `NautobotModelForm` definition at line 685/719 below.